app/src/main/java/com/GrowthPlus/ParentSignup.java

```diff
@@ -2,13 +2,129 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.Toast;
 
-public class ParentSignup extends AppCompatActivity {
+import com.GrowthPlus.dataAccessLayer.parent.ParentSchema;
+import com.GrowthPlus.dataAccessLayer.parent.ParentSchemaService;
+import com.GrowthPlus.realmImporter.JsonSampleData;
+
+import io.realm.Realm;
+import org.bson.types.ObjectId;
+
+public class ParentSignup extends AppCompatActivity implements View.OnClickListener{
+
+    Realm realm;
+    Resources resources;
+    private Button signupButton;
+    private Button signupBackButton;
+
+    private EditText enterPinInput;
+    private EditText confirmPinInput;
+    Integer enterPinInputInteger;
+    Integer confirmPinInputInteger;
+
+    private ObjectId parentId;
+    private String parentIdString;
+
+    private ParentSchemaService signupParentService;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_parent_signup);
+        init();
+
+        signupButton.setOnClickListener(this);
+        signupBackButton.setOnClickListener(this);
+
+    }
+
+    private void init(){
+        realm = Realm.getDefaultInstance();
+        resources = getResources();
+        signupButton = findViewById(R.id.parentSignupBtn);
+        signupBackButton = findViewById(R.id.backParentSU);
+        enterPinInput = findViewById(R.id.enterPin);
+        confirmPinInput = findViewById(R.id.confirmPinInput);
+        parentId = new ObjectId();
+        parentIdString = parentId.toString();
+    }
+
+    @Override
+    public void onClick(View view) {
+
+        int id = view.getId();
+
+        //if they click the signup button
+        if(id == R.id.parentSignupBtn){
+            //track what the user entered
```

Comment:
To address clicking the login/signup button with empty input, you can make a if statement 
`if (input.getText.toString() != "")` follow the logic you have written
`else` toast message saying empty input